### PR TITLE
Remove deprecation warning from `gr.update` and clean up associated code

### PR DESCRIPTION
**Pull request by abidlabs**
As suggested in #5863, this removes the deprecation warning from `gr.update()` as it can be handy in some situations, such as when you need to update the same property but across different output components. 

Also cleans up some old code that was causing warnings to be emitted incorrectly. I'm pretty sure its safe to delete, but would appreciate another pair of eyes.

Closes: #5863
Closes: #5746

### CODE REVIEW

**Comment by sumitsahoo** [comment]
Waiting to get this merged. Been receiving the warnings when using `llama-index`

```
UserWarning: Using the update method is deprecated. Simply return a new object instead, e.g. `return gr.Button(...)` instead of `return gr.Button.update(...)`.
```

File: gradio/blocks.py [hunk 1]
```diff
@@ -225,14 +225,6 @@ def get_config(self):
     def update(**kwargs) -> dict:
         return {}
 
-    @classmethod
```
Comment (freddyaboulton) [inline gradio/blocks.py:228]: Looks like this was added here: https://github.com/gradio-app/gradio/pull/2309

The repros in the issue still work so I think we're ok to delete.

**Review by freddyaboulton — APPROVED** [review]
Looks good to me @abidlabs !

**Review by aliabid94 — APPROVED** [review]
LGTM

**Comment by abidlabs** [comment]
Thanks folks for reviewing!

**Pull request merged**